docs/index.md

```diff
@@ -116,15 +116,24 @@ after execute_perform
 => true
 ```
 
-### Performer
+### Context & performer
 
-Every BA has a performer which can be assigned via a `.as` class method before BA creation.
+Every BA has a context which is a hash that can be assigned via `.using` class method before BA 
+initialization. Context is usually used to pass performer of the action which is so common that there
+are methods defined to access & set performer
+`performer` specifically.
 
 ```ruby
-MyAction.as(Admin.first).new(params)
-```
+action = MyAction.using(performer: Admin.first).new(params)
```

Comment:
Should we enforce symbols?

Reply:
I'm not sure. If we enforce anything I'd prefer to enforce using something that actually implements `performer` method, rather than just `[:performer]`, but something like `using(MyApplicationContext.new(current_user))`, doesn't look that nice.

Reply:
Maybe something like:
```ruby
class Granite::Context
  attr_reader :performer
  def initialize(performer)
    @performer = performer
  end
end

def using(*args, **options, &block)
  Proxy.new(self, Granite.context_class.new(*args, **options, &block))
end
```

So `Granite.context_class` would point to `Granite::Context` by default, but allow application to customize which class it wants to use. It adds a bit of complexity, but keeps the nice interface`BA.using(performer).new` while also allowing custom attributes in the context.